fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/Value.java

```diff
@@ -552,4 +552,34 @@ default List<Value> simplifyOrderingValue(@Nonnull final AliasMap aliasMap, @Non
             return ImmutableList.of(Simplification.simplify(simplifiedOrderingValue, aliasMap, constantAliases, OrderingValueSimplificationPerPartRuleSet.ofOrderingSimplificationPerPartRules()));
         }
     }
+
+    @SuppressWarnings("PMD.CompareObjectsWithEquals")
+    default boolean subsumedBy(@Nullable final Value other, @Nonnull final AliasMap aliasMap) {
+        if (other == null) {
```

Comment:
Can we employ the same scheme wrt to using a `subsumedWithoutChildren()` or does that not work? Alternatively, we may be abled to express this as a regular fold as `Value` implements `TreeLike<Value>`.